### ISSUE TITLE
`vg snarls -u` produces incorrect bubbles

### ISSUE THREAD

**Issue by adamnovak** [vgteam/vg#653]
Shilpa has a graph:

```
{"node": [{"sequence": "CCAAAATTTATTCAACTACTGAAGCTCATTAACGATGGTGGTAGTGATGACAGTGAATCTTCGTTGGCCACGGCTCAAAACATCTTGAACGAGGTCAACTTACCCTTATATAGAGAATACGACGATGACGTCAGTGAGATACTCGAGCAACTGTTCAATCGTATCAAATATTTGAGATTAGATGACGGTGGGCCCAAGCAAGGTCCAGTGACCGTCGACGTGCCCTTAACAGAGCCTTACTTTACGAGGTTCAAAGGAAAAGATGGTACTGATTATGCCCTCGCCAACAATGGCTGGATATGGAATGGTAACCCACTAGTGGATTTTGCATCGCAG", "id": 615}, {"sequence": "TGTCAATTCTGAACCCGTCAAATATCTTGGCGTTCATTTCTATATACTTGGACATTCTTTCCCACAGATACGGAGAGTCTTCAGGGCTTTTACCGTATCTTAACTTGACACAGTCCCCCCACACGATAACTTCTCTACGTAAATAAGCTCTCGAATTCTGCGATGCAAAATCCACTAGTGGGTTACCATTCCATATCCAGCCATTGTTGGCGAGGGCATAATCAGTACCATCTTTTCCTTTGAACCTCGTAAAGTAAGGCTCTGTTAAGGGCACGTCGACGGTCACTGGACCTTGCTTGGGC", "id": 7826}, {"sequence": "TATTTACGTAGAGAAGTTATCGTGTGGGGGGACTGTGTCAAGTTAAGATACGGTAAAAGCCCTGAAGACTCTCCGTATCTGTGGGAAAGAATGTCCAAGTATATAGAAATGAACGCCAAGATATTTGACGGGTTCAGAATTGACAACTGCCATTCTACTCCAATACATGTTGGCGAATATTTCCTAGATTTGGCAAGAAAATACAACCCGAACCTATATGTCGTTGC", "id": 42}, {"sequence": "GCCCAAGCAAGGTCCAGTGACCGTCGACGTGCCCTTAACAGAGCCTTACTTTACGAGGTTCAAAGGAAAAGATGGTACTGATTATGCCCTCGCCAACAATGGCTGGATATGGAATGGTAACCCACTAGTGGATTTTGCATCGCAGAATTCAAGAGCCTATTTACGTAGAGAAGTTATCGTGTGGGGGGACTGTGTCAAGTTAAGATACGGTAAAAGCCCTGAAGACTCTCCGTATCTGTGGGAAAGAATGTCCAAGTATATAGAAATGAACGCCAAGATATTTGACGGGTTCAGAATTGACAACT", "id": 83088}], "edge": [{"to_end": true, "from": 615, "to": 7826}, {"from": 615, "to": 83088}, {"from": 42, "to": 7826, "from_start": true}, {"from": 83088, "to": 42}]}
```

If you get the bubbles with `vg snarls -u`, you get two bubbles that don't really appear to be real:
```
vg view -Jv graph.json | vg snarls -u -

615	7826	615,7826,
83088	42	42,83088,
```

If you use normal `vg snarls`, you get the single ultrabubble that the graph actually is:

```
vg view -Jv graph.json | vg snarls - | vg view -Rj -

{"end": {"backward": true, "node_id": 615}, "type": 1, "start": {"backward": true, "node_id": 42}}
```

This looks to be because `vg::ultrabubbles()` in `graph.cpp` is returning nonsense, despite backending into the same code as the `SnarlFinder`.

Since `vg::ultrabubbles()` doesn't actually account for orientation in its returned nodes, I think we should just cut it ratehr than trying to fix it up when we have the new snarl system.

@glennhickey: Is there a good reason to keep `vg::ultrabubbles()` and the `-u` option on `vg snarls`?

**Comment by glennhickey** [comment]
The only use for the old code path that I can think of is for the tests,
where it's been handy to have the common interface between super and ultra
bubbles, and to verify the output was the same for dags using diff.

I'll take a look and see if there's an easy fix, but in general I agree
getting rid of it.

**Comment by glennhickey** [comment]
Fixed in #659
